Ffm 908

milos85vasic-harness opened the pull request:
This pull request contains the following changes:

- Metrics implementation for the Android (with basic unit tests)
- Feature configurations obtain mechanism
- Unit tests updates for easier future tests writing and extension
- Basic SDK documentation on how to use Metrics.

davejohnston reviewed (approved):
This looks good to me

rushabh-harness commented on README.md:
````diff
@@ -129,6 +129,29 @@ Following table provides summary on possible event types and corresponding respo
+In order to turn on metrics for feature flags evaluations it is required to define your configuration like in the following example:
+```kotlin
+val remoteConfiguration = CfConfiguration.builder()
+            .enableStream(true)
````
this should be default configuration, no need to define it explicitly

rushabh-harness commented on README.md:
```diff
@@ -129,6 +129,29 @@ Following table provides summary on possible event types and corresponding respo
+val remoteConfiguration = CfConfiguration.builder()
+            .enableStream(true)
+            .pollingInterval(10)
+            .enableAnalytics(true)
```
same thing for analytics

rushabh-harness commented on README.md:
````diff
@@ -129,6 +129,29 @@ Following table provides summary on possible event types and corresponding respo
+```kotlin
+val remoteConfiguration = CfConfiguration.builder()
+            .enableStream(true)
+            .pollingInterval(10)
````
what is the default polling interval acc to playbook ?

rushabh-harness commented on cfsdk/src/main/java/io/harness/cfsdk/cloud/model/Target.java:
```diff
@@ -1,14 +1,32 @@
+
     private String name;
     private String identifier;
+    private boolean isPrivate; // If the target is private
```
There is only 1 target in the system so not sure what this means if the target is private

rushabh-harness commented on cfsdk/src/main/java/io/harness/cfsdk/cloud/model/Target.java:
```diff
@@ -1,14 +1,32 @@
     private String name;
     private String identifier;
+    private boolean isPrivate; // If the target is private
+    private Set<String> privateAttributes; // Custom set to set the attributes which are private
```
Does it make sense to have private attributes in client SDK ?

rushabh-harness reviewed (changes requested):
Please go over the comments I have listed in the PR. Let me know what you feel

milos85vasic-harness commented:
@rushabh-harness I have addressed all of your comments since each of them Is valid. Please look at my latest changes (my last commit) and let me know once you finish the review. Thank you!
